server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java

```diff
@@ -194,8 +194,14 @@ public enum MetadataVersion {
     // Add JBOD support for KRaft.
     IBP_3_7_IV2(17, "3.7", "IV2", true),
 
+    // IBP_3_7_IV3 was ELR related supports (KIP-966) and has been moved forward
+    IBP_3_7_IV3(18, "3.7", "IV3", false),
+
+    // Add new fetch request version for KIP-951
+    IBP_3_7_IV4(19, "3.7", "IV4", false),
+
     // Add ELR related supports (KIP-966).
-    IBP_3_7_IV3(18, "3.7", "IV3", true);
+    IBP_3_8_IV0(20, "3.8", "IV0", true);
```

Comment:
Can you split this change into two PRs/commits? One that fixes the MV in 3.7 and one that introduced the new 3.8 MV. The one that fixes the 3.7 versions should be cherry picked to the 3.7 branch.

Reply:
If we remove the tag from here should we also remove all the other code associated with ELR from the 3.7 branch?

Reply:
You just need to return false for `isElrSupported()`. Is that what you are asking?

Reply:
No, you asked to split this PR into two separate PRs, one for 3.7 without any reference to 3.8 and this one. I am saying that doing that has no value because 3.8 is labeled as unstable so having it in 3.7 isn't an issue. Customers cannot use 3.8 unless they explicitly allow unstable metadata versions.